go-controller/pkg/clustermanager/clustermanager.go

```diff
@@ -0,0 +1,271 @@
+		defaultNetClusterController: newDefaultNetworkClusterController(ovnClient, wf),
+		wg:                          wg,
+		wf:                          wf,
+		stopChan:                    make(chan struct{}),
+		recorder:                    recorder,
+	}
+
+	if config.OVNKubernetesFeature.EnableMultiNetwork {
+		cm.secondaryNetClusterControllerManager = newSecondaryNetworkClusterControllerManager(ovnClient, wf, recorder)
+	}
+	return cm
+}
+
+// Start waits until this process is the leader before starting master functions
+func (cm *ClusterManager) Start(nodeName string, ctx context.Context) error {
+	klog.Infof("Cluster manager Started")
+	metrics.RegisterClusterManagerBase()
+
+	// Set up leader election process first.
+	// User lease resource lock as configmap and endpoint lock support is removed from leader election library.
+	rl, err := resourcelock.New(
+		resourcelock.LeasesResourceLock,
+		config.Kubernetes.OVNConfigNamespace,
+		"ovn-kubernetes-cluster-manager",
+		cm.client.CoreV1(),
+		cm.client.CoordinationV1(),
+		resourcelock.ResourceLockConfig{
+			Identity:      nodeName,
+			EventRecorder: cm.recorder,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	lec := leaderelection.LeaderElectionConfig{
+		Lock:            rl,
+		LeaseDuration:   time.Duration(config.ClusterMgrHA.ElectionLeaseDuration) * time.Second,
+		RenewDeadline:   time.Duration(config.ClusterMgrHA.ElectionRenewDeadline) * time.Second,
+		RetryPeriod:     time.Duration(config.ClusterMgrHA.ElectionRetryPeriod) * time.Second,
+		ReleaseOnCancel: true,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {
+				klog.Infof("Won leader election; in active mode")
+				// run the cluster controller to init the cluster manager
+				start := time.Now()
+				defer func() {
+					end := time.Since(start)
+					metrics.MetricClusterManagerReadyDuration.Set(end.Seconds())
+				}()
+
+				// run only on the active master node.
+				if err := cm.Run(); err != nil {
+					panic(err.Error())
+				}
+			},
+			OnStoppedLeading: func() {
+				// This node was leader and it lost the election.
+				// Whenever the node transitions from leader to follower,
+				// we need to handle the transition properly like clearing
+				// the cache. It is better to exit for now.
+				// kube will restart and this will become a follower.
+				klog.Infof("No longer leader; exiting")
+				os.Exit(0)
+			},
+			OnNewLeader: func(newLeaderName string) {
+				if newLeaderName != nodeName {
+					klog.Infof("Lost the election to %s; in standby mode", newLeaderName)
+				}
+			},
+		},
+	}
+
+	leaderelection.SetProvider(ovnkubeClusterManagerLeaderMetricsProvider{})
+	leaderElector, err := leaderelection.NewLeaderElector(lec)
+	if err != nil {
+		return err
+	}
+
+	cm.wg.Add(1)
+	go func() {
+		leaderElector.Run(ctx)
+		klog.Infof("Stopped leader election")
+		cm.wg.Done()
+	}()
+
+	return nil
+}
+
+func (cm *ClusterManager) Stop() {
```

Comment:
you dont need to expose a `Stop` fn. If users want to stop this, they can stop the stop chan passed into `Start`.

Reply:
There's only Start and Stop public functions now.